### ISSUE TITLE
Whole HTML page can be unexpectedly requested multiple times

### ISSUE THREAD

**Issue by andykent** [imgix/react-imgix#49]
There is a pretty epic and hard to spot bug in ReactImgix. It took me several hours to track this down as it's very subtle.

We were observing multiple requests to our backend servers for each pageview that included ReactImgix on it. It turns out that there is a subtle browser behaviour where a blank URL is interpreted as 'the current page' this means setting `backgroundImage = url()` will actually attempt to request the entire html page as an image. I believe this is also true of `<img src="" />` nodes too.

The two specific bugs in ReactImgix are...
- setting `bg=true` causes the initial render pass to generate styles such as `backgroundImage = url()`
- without `aggressiveLoad` set the initial render pass generates an img tag like `<img src="" srcset="" />`

It's particularly noticeable when using server side rendering as this first render pass is what gets returned to the client because there is no DOM to mount. 

For the `bg=true` case I think this can be easily fixed by simply not setting the `backgroundImage` style attr until the src attr is computed and present. E.g. we have patched this locally using...

```
if (_src && _src.length > 0) childProps.style.backgroundImage = `url(${_src})`
```

Some possible, but as yet totally untested solutions for the `src` attr might be to...
- leave the src attribute off entirely (this would be best if it works but I've not tested it)
- set `src="about:blank"` (smallest change, not sure if it would work)
- render a `<span>` tag instead of the actual image. (probably a breaking change and not really desirable but would work)

**Comment by paulstraw** [comment]
Hey @andykent, thanks so much for this thorough report. It sounds like you've already got one fix in place (which looks sensible to me), and have some good theories on the second fix. Would you like to submit a PR for this? If not, I'm totally happy to work on it, but I just wanted to check since it seems like you've got a really good handle on the problem.

**Comment by frederickfogerty** [comment]
😱  The new things you learn every day!

Thanks for the bug report @andykent, it seems like you have a pretty firm grasp on the problem. I think the best approach would be to work through the possible solutions you have outlined in order. Is the patch you outlined in user-land, or did you modify the library source? If you have any code that would fix this issue in the source, please feel free to submit it as a PR and we can pick it up. If not, no worries, one of us will get onto fixing it.

**Comment by andykent** [comment]
We actually fix it in userland by sub-classing the component right now. I can try and have a go at a PR tomorrow though if I get a few mins to spare.

**Comment by andykent** [comment]
Fixed in #51

**Comment by frederickfogerty** [comment]
I just wanted to comment here as I've changed the solution to this issue in order to help #41. With no `src` tag, an `<img />` is invalid, and behaves unexpectedly in some browsers (renders a `0 x line-height` box in FF). I've updated the src to be `//:0`, which is an invalid url (port can't be 0), but it makes the image render properly. More information can be found [here](https://stackoverflow.com/questions/19126185/setting-an-image-src-to-empty/28486885) and [here](https://stackoverflow.com/a/5775621/515634)
